### PR TITLE
feat: 🍰Implement Registration Slider

### PR DESCRIPTION
**Pull request by Tirokk**
## 🍰 Pullrequest

Implement registration with slider.

Link for:

- invite code: `http://localhost:3000/registration?method=invite-code&inviteCode=AAAAAA`
- invite e-mail: `http://localhost:3000/registration?method=invite-mail&email=wolle.huss%40pjannto.com&nonce=64853`

### Issues

- relates #4092

### Test manually

Public registration

- Cypher for query nonce by e-mail: `MATCH (email:EmailAddress {email: "wolle.huss@pjannto.com"}) RETURN email;`

Should be tested, because of partial code connections:

- change e-mail
  - Cypher for query nonce for change: `MATCH (email:UnverifiedEmailAddress {email: "github@pjannto.com"}) RETURN email;`
- change password
- admin invite
  - Cypher for query nonce by e-mail: `MATCH (email:EmailAddress {email: "wolle.huss@pjannto.com"}) RETURN email;`

### Todo

- frontend
  - [x] change nonce length to 5
  - [x] implement new nonce query
  - [x] slider: public registration not possible
  - [x] invent env `INVITE_REGISTRATION`
  - [x] removed about you
  - [x] remove some agreements which the user has to agree to
  - [x] translate `No Public Registration`
  - [x] `webapp/components/EnterNonce/EnterNonce.vue`
    - locale `enter-invite.form.validations` mentions 6 characters …
  - [x] `webapp/components/EnterNonce/EnterNonce.vue`
    - [x] show example code
    - locale `enter-nonce.form.validations` mentions 6 characters …
    - [x] show example code
    - [x] should be 5
  - [x] have toaster for wrong invite code and wrong nonce
  - [x] remove main headline and put it on the first slider
  - [x] delete all old registration possibilities:
    - [x] file: `webapp/pages/registration/create-user-account.vue`
    - [x] file: `webapp/pages/registration/enter-nonce.vue`
    - [x] file: `webapp/pages/registration/signup.vue`
    - [x] file: `webapp/components/Registration/CreateUserAccount.spec.js`
    - [x] file: `webapp/components/Registration/CreateUserAccount.story.js`
    - [x] file: `webapp/components/Registration/CreateUserAccount.vue`
   - [x] more files … ?
   - [x] delete locales as well
     - [x] `termsAndConditions.termsAndConditionsConfirmed`
     - [x] `components.registration.signup.form.data-privacy`
     - [x] `components.registration.signup.form.minimum-age`
     - [x] `components.registration.signup.form.no-commercial`
     - [x] `components.registration.signup.form.no-political`
  - keep files, because they are used:
    - used in password-reset `webapp/pages/password-reset/enter-nonce.vue`:
      - file: `webapp/components/EnterNonce/EnterNonce.vue`
    - used in admin/invite `webapp/pages/admin/invite.vue`:
      - file: `webapp/components/Registration/Signup.spec.js`
      - file: `webapp/components/Registration/Signup.vue`
  - [x] (create issue) write frontend tests (see issue #4311 )
  - [x] (create issue) use make password visible like in LoginForm? (see issue #4312 )
- backend
  - [x] invent env `INVITE_REGISTRATION`
  - [x] implement test for mutation `SignupVerification` with `inviteCode` (see issue #4313 )
  - [ ] fix backend test in `slugifyMiddleware`
    - seems a flaky test?
- e-mails
  - [x] change admin e-mail invite to `http://localhost:3000/registration?method=invite-mail&email=wolle.huss%40pjannto.com&nonce=64853`
  - [x] change public registration e-mail to `http://localhost:3000/registration?method=invite-mail&email=wolle.huss%40pjannto.com&nonce=64853`
  - [x] check e-mail content is the `nonce` visible enough?
- general
  - [x] remove `SignupByInvitation`
 - deployment
  - [x] invent env `INVITE_REGISTRATION`

### CODE REVIEW

**Comment by Mogge** [comment]
I added a `VerifyNonce(email: String!, nonce: String!)` query which returns a boolean. @Tirokk

**Comment by Tirokk** [comment]
Hey nice @Mogge ! Then I will proceed with it now.

**Comment by Mogge** [comment]
@Tirokk I added ther permissions for Signup when using invite code. Next I will add the SignupVerification

**Comment by Tirokk** [comment]
Great @Mogge ! 😎 I paused today what was really good 👍🏼

File: webapp/.env.template [hunk 1]
```diff
@@ -1,5 +1,6 @@
 SENTRY_DSN_WEBAPP=
 COMMIT=
 PUBLIC_REGISTRATION=false
+INVITE_REGISTRATION=true
```
Comment (Mogge) [inline webapp/.env.template:4]: Do we really want this set to `true` in the template? I think it is better to set it to `false`

File: backend/.env.template [hunk 2]
```diff
@@ -17,6 +17,7 @@ PRIVATE_KEY_PASSPHRASE="a7dsf78sadg87ad87sfagsadg78"
 SENTRY_DSN_BACKEND=
 COMMIT=
 PUBLIC_REGISTRATION=false
+INVITE_REGISTRATION=true
```
Comment (Mogge) [inline backend/.env.template:20]: Do we really want this set to true in the template? I think it is better to set it to false

**Review by Mogge — COMMENTED** [review]
I have doubts about the default values in the env template

**Comment by Mogge** [comment]
@Tirokk So why do you answer when you take a day off? Please try to really disconnect when you are not working :)

**Comment by Tirokk** [comment]
Hey @Mogge , `yarn test components/Registration/Signup.spec.js` fails at the moment.

And in the `Signup` mutation the `inviteCode` is not used at the moment?

**Comment by ulfgebhardt** [comment]
https://github.com/Ocelot-Social-Community/Ocelot-Social-Deploy-Rebranding/commit/874d719ae32731f908bb2c0367c535609d6f3004

I support `INVITE_REGISTRATION` in the deployment now. I choose default true - there was a discussion above I read after. Is `INVITE_REGISTRATION: true` incorrect for a default?

**Comment by ulfgebhardt** [comment]
If I run the branch locally I get a looping redirect when opening localhost:3000 - is this an old redirect or what happening?
i redirect between:

image

and

image

-> The URL stays the same: `http://localhost:3000/registration/signup?path=%2Fregistration%2Fsignup`

-> After around 10 redirects the server does not respond anymore (or the browser for that matter.) Processor is not in use.

-> This is run in a fresh full-docker environment.

**Comment by ulfgebhardt** [comment]
The redirect thing seems fixed after the commits you pushed @Tirokk 
I still have a broken URL tho.

**Comment by Tirokk** [comment]
Hey @Mogge ,
I have a problem with the combination of `Signup` mutation, `VerifyNonce`, and `SignupVerification` mutation query.

- `Signup` creates a `email:EmailAddress` node.
- `VerifyNonce` checks if a `email:UnverifiedEmailAddress` node exist. Which does not match the first.
- And `SignupVerification` again handles an `email:EmailAddress` node.

I can correct it my self, but because I don’t know your deeper thoughts behind `email:UnverifiedEmailAddress` instead of `email:EmailAddress` I would prefer you having a look on it.

**Comment by Mogge** [comment]
@Tirokk I did not do the `UnverifiedEmailAddress`. I think it is needed when you change your Email Address.

**Review by ulfgebhardt — APPROVED** [review]

**Pull request merged**